model/fv_dynamics.F90

```diff
@@ -849,7 +862,44 @@ subroutine fv_dynamics(npx, npy, npz, nq_tot,  ng, bdt, consv_te, fill,
        used = send_data(idiag%id_mdt, dtdt_m, fv_time)
   endif
 
-  if( nwat==6 ) then
+  if( nwat==7 ) then
```

Comment:
This is the right way to add a new nwat to the code. I am hoping that eventually there can be a more generic way to do this though so that we don't need separate hard codings for different numbers of ice categories.

Reply:
That would work. There should be an easy way for the physics to query the dynamics for the nwat value.

Reply:
@bensonr Right, nwat is being passed to model%init as intent(in). It looks like it is used before that, so it might take some changes to be able to set it in model%init. Something to look at for a future update?